bcipy/signal/evaluate/artifact.py

```diff
@@ -0,0 +1,595 @@
+
+    The artifact type enum is used to define the types of artifacts that can be detected in the data.
+    """
+
+    VOLTAGE = 'voltage'
+    EOG = 'eog'
+    BLINK = 'blink'
+    ECG = 'ecg'
+    EMG = 'emg'
+    FLAT = 'flat'
+    EVENT = 'event'
+
+    def __str__(self) -> str:
+        """Return the value of the artifact type."""
+        return self.value
+
+    def __repr__(self) -> str:
+        """Return the class name and the artifact type name."""
+        return f'{self.__class__.__name__}.{self.name}'
+
+    def label(self) -> str:
+        """Label the artifact type.
+
+        Returns
+        -------
+        label - string - The label for the artifact type with the prefix BAD.
+                         This is used for epoch handling by MNE.
+        """
+        return f'BAD_{self.value}'
+
+
+class ArtifactDetection:
+    """Artifact Detection Class.
+
+    The artifact detection class is used to detect and label artifacts in the raw data. This class is designed to be
+    used in a pipeline for artifact detection and labelling.
+
+
+    Parameters
+    ----------
+    raw_data : RawData
+        The raw data to process for artifacts.
+
+    parameters : dict
+        The parameters used in the collection of the raw data.
+
+    device_spec : dict
+        The device specification for the raw data.
+
+    save_path : optional str
+        The path to save the artifact labelled data. If none provided, data is not saved.
+
+    session_triggers : optional list
+        Optional list of session triggers to use for co-labelling the data. This is useful for determining the impact
+        of artifacts on the relevant epochs.
+
+    percent_bad : float
+        The percentage of bad epochs to allow before channel exclusion. Defaults to 50%.
+
+    detect_eog : bool
+        Whether to detect EOG artifacts. Defaults to True. Eye channels must be provided if this is set to True.
+
+    eye_channels : list
+        The list of eye channels to use for detecting EOG artifacts. Defaults to None. If provided, the channels should
+        look like ['Fp1', 'Fp2', 'F7', 'F8'].
+
+    detect_voltage : bool
+        Whether to detect voltage artifacts. Defaults to True.
+    """
+
+    supported_units: List[str] = ['volts', 'microvolts']
+    support_device_types: List[str] = ['EEG']
+    analysis_done: bool = False
+    dropped: Optional[str] = None
+    eog_annotations: Optional[Annotations] = None
+    voltage_annotations: Optional[Annotations] = None
+
+    def __init__(
+            self,
+            raw_data: RawData,
+            parameters: dict,
+            device_spec: DeviceSpec,
+            save_path: Optional[str] = None,
+            semi_automatic: bool = False,
+            session_triggers: Optional[Tuple[list, list, list]] = None,
+            percent_bad: float = 50.0,
+            detect_eog: bool = True,
+            eye_channels: Optional[List[str]] = None,
+            detect_voltage: bool = True) -> None:
```

Comment:
Rather than having a of boolean flag for each artifact type it might be more flexible here to have a single parameter that is a list of ArtifactTypes of interest.